src/replication.c

```diff
@@ -3684,13 +3684,18 @@ void syncWithPrimary(connection *conn) {
 
     /* Prepare a suitable temp file for bulk transfer */
     if (!useDisklessLoad()) {
+        /* We save the errno of open to prevent some systems from modifying it after
+         * the sleep call. For example, sleep in Mac will change errno to ETIMEDOUT. */
+        int saved_errno;
         while (maxtries--) {
             snprintf(tmpfile, 256, "temp-%d.%ld.rdb", (int)server.unixtime, (long int)getpid());
             dfd = open(tmpfile, O_CREAT | O_WRONLY | O_EXCL, 0644);
             if (dfd != -1) break;
+            saved_errno = errno;
             sleep(1);
```

Comment:
Another possibility is to save use the saved errno only during sleep. Smaller scope is simpler code?

```suggestion
            /* Bla bla bla MacOS bla bla ETIMEDOUT. */
            int saved_errno = errno;
            sleep(1);
            errno = save_errno;
```